packages/template-retail-react-app/app/hooks/use-dnt-modal.js

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {useEffect} from 'react'
+import PropTypes from 'prop-types'
+import {FormattedMessage, useIntl} from 'react-intl'
+import {
+    Button,
+    Modal,
+    ModalBody,
+    ModalCloseButton,
+    ModalContent,
+    useDisclosure,
+    Heading,
+    Stack,
+    Text,
+    Flex
+} from '@salesforce/retail-react-app/app/components/shared/ui'
+import {HideOnDesktop, HideOnMobile} from '@salesforce/retail-react-app/app/components/responsive'
+import {useDNT} from '@salesforce/commerce-sdk-react'
+import {useLocation} from 'react-router-dom'
+
+export const DntModal = ({isOpen, onOpen, onClose, ...props}) => {
+    const {dntStatus, updateDNT} = useDNT()
+    const {formatMessage} = useIntl()
+    const location = useLocation()
+
+    useEffect(() => {
+        if (dntStatus === undefined) onOpen()
+    }, [location])
+
+    const onCloseModal = () => {
+        updateDNT(null)
+        onClose()
+    }
+
+    const buttons = (
+        <>
+            <Button
```

Comment:
Can you add aria-labels on these button? Something like aria-labels={intl.formatMessage({id: 'dnw_modal.assistive_msg.defline', defaultMessage: "Decline Do not Track"})}